cmd/cli_test.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/NethermindEth/1click/configs"
+	"github.com/NethermindEth/1click/internal/pkg/commands"
+	"github.com/NethermindEth/1click/test"
+	log "github.com/sirupsen/logrus"
+)
+
+type cliCmdTestCase struct {
+	configPath     string
+	generationPath string
+	runner         commands.CommandRunner
+	fdOut          *bytes.Buffer
+	args           []string
+	isErr          bool
+}
+
+func resetCliCmd() {
+	cfgFile = ""
+	executionName = ""
+	consensusName = ""
+	validatorName = ""
+	generationPath = configs.DefaultDockerComposeScriptsPath
+	randomize = false
+	install = false
+	run = false
+	y = false
+	services = &[]string{}
+}
+
+func buildCliTestCase(t *testing.T, caseName string, args []string, isErr bool) *cliCmdTestCase {
+	tc := cliCmdTestCase{}
+	configPath := t.TempDir()
+
+	err := test.PrepareTestCaseDir(filepath.Join("testdata", "cli_tests", caseName, "config"), configPath)
+	if err != nil {
+		t.Fatalf("Can't build test case: %v", err)
+	}
+	dcPath := filepath.Join(configPath, "docker-compose-scripts")
+	err = os.Mkdir(dcPath, os.ModePerm)
+	if err != nil {
+		t.Fatalf("Can't build test case: %v", err)
+	}
+
+	// TODO: allow runner edition
+	tc.runner = &test.SimpleCMDRunner{
+		SRunCMD: func(c commands.Command) (string, error) {
+			return "", nil
+		},
+		SRunBash: func(bs commands.BashScript) (string, error) {
+			return "", nil
+		},
+	}
+
+	tc.args = args
+	tc.generationPath = dcPath
+	tc.configPath = filepath.Join(configPath, "config.yaml")
+	tc.fdOut = new(bytes.Buffer)
+	tc.isErr = isErr
+	return &tc
+}
+
+func TestCliCmd(t *testing.T) {
+	//TODO: allow to test error programs
+	tcs := []cliCmdTestCase{
+		*buildCliTestCase(t, "case_1", []string{"-r"}, false),
+		*buildCliTestCase(t, "case_1", []string{"-e", "nethermind", "-c", "lighthouse", "-v", "lighthouse"}, false),
```

Comment:
```suggestion
		*buildCliTestCase(t, "case_2", []string{"-e", "nethermind", "-c", "lighthouse", "-v", "lighthouse"}, false),
```

Reply:
if you check posmoni tests, you could agree on the fact that adding a description to the test case is a good iddea. Test cases should be self documenting. Think about how hard is for someone to read these tests and see what is going on, and what is the behavior each test case tests.

Reply:
Yes, I will add this.